src/text_file_functions.cpp

```diff
@@ -316,7 +317,80 @@ QString AOApplication::get_stylesheet(QString target_tag, QString p_file)
   return f_text; // This is the empty string if no appends took place
 }
 
-QVector<QStringList> AOApplication::get_highlight_color()
+QMap<Color, QStringList> AOApplication::get_chatmessage_colors()
+{
+  QMap<Color, QStringList> default_colors;
+  default_colors[CWhite] = QStringList{"White", "#D5D5D5"};
+  default_colors[CGreen] = QStringList{"Green", "#65C856"};
+  default_colors[CRed] = QStringList{"Red", "#BA1518"};
+  default_colors[COrange] = QStringList{"Orange", "#D55900"};
+  default_colors[CBlue] = QStringList{"Blue", "#1588C8"};
+  default_colors[CYellow] = QStringList{"Yellow", "#E7CE4E"};
+  default_colors[CPurple] = QStringList{"Purple", "#F776FD"};
+  default_colors[CPink] = QStringList{"Pink", "#DA7C80"};
+
+  // File lookup order
+  // 1. In the theme folder (gamemode-timeofday/main/default), look for
+  // "courtroom_text_colors.ini".
+
+  QString path = find_theme_asset_path("courtroom_text_colors.ini");
+  if (path.isEmpty())
+    return default_colors;
+
+  QSettings text_colors(path, QSettings::IniFormat);
+
+  // Make copy for new colors
+  QMap<Color, QStringList> read_colors;
+  foreach (Color color, default_colors.keys())
+  {
+    QStringList color_parameters = default_colors[color];
+    read_colors[color] = color_parameters;
+  }
+
+  foreach (QString color_name, text_colors.childGroups())
+  {
+    // Capitalize because inis may not necessarily capitalize, but DRO does
+    color_name = color_name[0].toUpper() + color_name.mid(1).toLower();
+
+    // Ignore colors that are not default colors
+    // Check if color is valid
+    Color matched_default_color;
+    bool valid = false;
+
+    foreach (Color default_color, default_colors.keys())
+    {
+      QStringList default_color_parameters = default_colors[default_color];
+      if (default_color_parameters[0] == color_name)
+      {
+        matched_default_color = default_color;
+        valid = true;
+        break;
+      }
+    }
+
+    if (!valid)
+      continue;
+
+    // At this point we know the color is valid
+    text_colors.beginGroup(color_name);
+    QString color_code = text_colors.value("code").toString();
+    if (color_code.isEmpty())
+      continue;
+    // Check if the color is valid. If not, set it to white
+    if (!QColor::isValidColor(color_code))
+    {
+      color_code = default_colors[CWhite][1];
+    }
```

Comment:
If the color isn't valid we shouldn't even consider adding it, why pick from `CWhite` when you could just make it so that we don't even consider this color to be valid?

Reply:
Furthermore, if a color isn't valid we should warn the user in the console.